AK/CharacterTypes.h

```diff
@@ -0,0 +1,178 @@
+/*
+ * Copyright (c) 2021, Max Wipfli <mail@maxwipfli.ch>
+ *
+ * SPDX-License-Identifier: BSD-2-Clause
+ */
+
+#pragma once
+
+#include <AK/Types.h>
+
+// NOTE: For a quick reference for most of this, see https://www.cplusplus.com/reference/cctype/ and https://infra.spec.whatwg.org/#code-points.
+// NOTE: To avoid ambiguity when including this header, all methods contains names should contain "ascii" or "unicode".
+
+#define DECLARE_BOOL_METHOD(name) \
+    constexpr bool name(u32 code_point)
+
+#define DECLARE_VALUE_METHOD(name) \
+    constexpr u32 name(u32 code_point)
+
+namespace AK {
+
+DECLARE_BOOL_METHOD(is_ascii)
```

Comment:
I don't personally see the value of the macro here... it just obfuscates the code IMHO.

Reply:
Removing this just leads to lots of code duplication. I've seen lots of macros to avoid code duplication all over the project (see e.g. `HTMLTokenizer`), so I thought that this is "the way it is done" here.

Reply:
Agree with @bgianfo, please remove these macros as they just make this harder to read.